src/tests/ftest/util/dmg_utils.py

```
@@ -23,44 +23,183 @@
+        if self.action.value == "format":
+            self.action_command = self.DmgFormatSubCommand()
+        elif self.action.value == "prepare":
+            self.action_command = self.DmgPrepareSubCommand()
+
+    class DmgFormatSubCommand(CommandWithParameters):
+        """Defines an object representing a format sub dmg command."""
+
+        def __init__(self):
+            """Create a dmg Command object."""
+            super(DmgCommand.DmgFormatSubCommand, self).__init__(
+                "/run/dmg/format/*", "format")
+            self.force = FormattedParameter("-f", False)
+
+    class DmgPrepareSubCommand(CommandWithParameters):
+        """Defines a object representing a prepare sub dmg command."""
+
+        def __init__(self):
+            """Create a dmg Command object."""
+            super(DmgCommand.DmgPrepareSubCommand, self).__init__(
+                "/run/dmg/prepare/*", "prepare")
+            self.pci_wl = FormattedParameter("-w {}")
+            self.hugepages = FormattedParameter("-p {}")
+            self.targetuser = FormattedParameter("-u {}")
+            self.nvmeonly = FormattedParameter("-n", False)
+            self.scmonly = FormattedParameter("-s", False)
+            self.force = FormattedParameter("-f", False)
+            self.reset = FormattedParameter("--reset", False)
+
+
+def storage_scan(hosts):
+    """ Execute scan command through dmg tool to servers provided.
+
+    Args:
+        hosts (list): list of servers to run scan on.
+
+    Returns:
+        Avocado CmdResult object that contains exit status, stdout information.
+
+    """
+    # Create and setup the command
+    dmg = DmgCommand(get_file_path("bin/daos_shell"))
+    dmg.request.value = "storage"
+    dmg.action.value = "scan"
+    dmg.insecure.value = True
+    dmg.hostlist.value = hosts
+
+    try:
+        result = dmg.run()
+    except CommandFailure as details:
+        print("<dmg> command failed: {}".format(details))
+        return False
```

To avoid returning different object types, this should either raise an exception at this point or return `None`.  This would indicate that no devices were found due to an error (None returned) vs. no devices where found with a successful command (empty list returned).

---

Changed to return None.